### PR TITLE
Get master

### PR DESCRIPTION
**Pull request by cyrillbolliger**
#24 zuerst anschauen (die commits davon sind hier schon drin)

### CODE REVIEW

File: app/Repository/Member/MasterDetector.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		}
+
+		return $master;
+	}
+
+	/**
+	 * Look at the five member fields and value the 'member' status with 11, 'unconfirmed' with 6 and
+	 * 'sympathizer' with 1.
+	 *
+	 * The weights of 11, 6 and 1 assure that one member status always wins over five sympathizer status
```
Comment (AScheuss) [inline app/Repository/Member/MasterDetector.php:93]: Vielleicht könnte man noch schreiben, dass fünf gewählt wurde, weil das die "grösste Tiefe" ist (Bund, Kanton, Bezirk, Gemeinde, Junge Grüne)

File: app/Repository/Member/Member.php [hunk 2]
```diff
@@ -231,33 +231,52 @@ public function __construct(
...
 	}
-	
+
+	/**
+	 * Make sure the members properties get cloned as well
+	 */
+	public function __clone() {
+		$fields = get_object_vars( $this );
+		foreach ( $fields as $fieldKey => $fieldValue ) {
+			if ( is_object( $fieldValue ) ) {
```
Comment (AScheuss) [inline app/Repository/Member/Member.php:241]: Es sind nur Objekte oder Arrays möglich (oder relevant) auf dieser Ebene und auch für einen Array gibt es nicht einen Eintrag mit Array?
I.e. Die verbliebenen Corner-Cases sind nicht wichtig, nehme ich an.

**Review by AScheuss — APPROVED** [review]
Sieht gut aus. Danke.
(Die Tests habe ich nicht mehr so genau angeschaut.)

File: app/Repository/Member/MasterDetector.php (hunk 1 above)
Comment (cyrillbolliger) [reply app/Repository/Member/MasterDetector.php:93]: Guter Punkt. Habe ich ergänzt.

File: app/Repository/Member/Member.php (hunk 2 above)
Comment (cyrillbolliger) [reply app/Repository/Member/Member.php:241]: Die primitiven Datentypen werden von PHPs default Verhalten bereits kopiert. Von den komplexen Datenstrukturen macht PHP jedoch nur eine shallow copy. 

Mit dieser Methode, forcieren wir eine echte Kopie der komplexen Datentypen (nur auf einem Level, keine deep copy). Für unsere Anwendung sollte das eigentlich ausreichen. Sobald wir aber grundsätzliche Änderungen am Memberobjekt machen, müssen wir die `__clone` Methode anpassen. Da es sich hier um klasseninterne Änderungen handelt, finde ich diese Vereinfachung zulässig.
